Src/GoogleApis.Auth.DotNet4/OAuth2/GoogleCredential.cs

```diff
@@ -0,0 +1,307 @@
+        /// <para>Returns the Application Default Credentials which are ambient credentials that identify and authorize 
+        /// the whole application.</para>
+        /// <para>The ambient credentials are determined as following order:</para>
+        /// <list type="number">
+        /// <item> 
+        /// <description>The environment variable GOOGLE_APPLICATION_CREDENTIALS is checked. If this variable is specified, it 
+        /// should point to a file that defines the credentials. The simplest way to get a credential for this purpose is to 
+        /// create a service account using the <a href="https://console.developers.google.com">Google Developers Console</a> in 
+        /// the section APIs &amp; Auth, in the sub-section Credentials. Create a service account or choose an existing one and 
+        /// select Generate new JSON key. Set the environment variable to the path of the JSON file downloaded.</description> 
+        /// </item> 
+        /// <item> 
+        /// <description>If you have installed the Google Cloud SDK on your machine and have run the command 
+        /// <a href="https://cloud.google.com/sdk/gcloud/reference/auth/login">gcloud auth login</a>, your identity can be used as 
+        /// a proxy to test code calling APIs from that machine.</description> 
+        /// </item> 
+        /// <item> 
+        /// <description>If you are running in Google Compute Engine production, the built-in service account associated with the 
+        /// virtual machine instance will be used.</description> 
+        /// </item> 
+        /// </list>
+        /// </summary>
+        public static Task<GoogleCredential> GetApplicationDefaultAsync()
+        {
+            return defaultCredentialProvider.GetDefaultCredentialAsync();
+        }
+
+        /// <summary>
+        /// Load credential from stream containing JSON credential data.
+        /// <para>
+        /// The stream can contain a Service Account key file in JSON format from the Google Developers
+        /// Console or a stored user credential using the format supported by the Cloud SDK.
+        /// </para>
+        /// </summary>
+        public static GoogleCredential FromStream(Stream stream)
+        {
+            return defaultCredentialProvider.CreateDefaultCredentialFromStream(stream);
+        }
+
+        /// <summary>
+        /// <para>Returns true only if this credential type has no scopes by default and requires 
+        /// a call to createScoped before use.</para>
+        ///
+        /// <para>Credentials need to have scopes in them before they can be used to access Google services. 
+        /// Some Credential types have scopes built-in, and some dont. This property indicates whether 
+        /// the Credential type has scopes built-in.</para>
+        /// 
+        /// <list type="number">
+        /// <item> 
+        /// <description>ComputeCredential has scopes built-in. Nothing additional is required.</description> 
```

Comment:
Add links to ComputeCredential, UserCredential or whatever credential you mention